### PR TITLE
feat(nat): add a new datasource to query the list of private nat gateway

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:
Add a new datasource to query the list of private nat gateway

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Add a new datasource
1.1 the datasource is used to query the list of private nat gateway
1.2 add the datasource documentation (docs/datasource/nat_private_gateways)
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud# make testacc TEST="./huaweicloud/services/acceptance/nat" TESTARGS="-run TestAccDatasourcePrivateGateways_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/nat -v -run TestAccDatasourcePrivateGateways_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourcePrivateGateways_basic
=== PAUSE TestAccDatasourcePrivateGateways_basic
=== CONT  TestAccDatasourcePrivateGateways_basic
--- PASS: TestAccDatasourcePrivateGateways_basic (102.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/nat       102.661s
```

### CODE REVIEW

File: docs/data-sources/nat_private_gateways.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+data "huaweicloud_nat_private_gateways" "test" {
+  name = var.name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region where the private NAT gateway is located.
```
Comment (Lance52259) [inline docs/data-sources/nat_private_gateways.md]:
```suggestion
* `region` - (Optional, String) Specifies the region where the private NAT gateways are located.  
```

File: docs/data-sources/nat_private_gateways.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region where the private NAT gateway is located.  
+  If omitted, the provider-level region will be used.
+
+* `gateway_id` - (Optional, String) Specifies the ID of the private NAT gateway instance.
```
Comment (Lance52259) [inline docs/data-sources/nat_private_gateways.md]:
```suggestion
* `gateway_id` - (Optional, String) Specifies the ID of the private NAT gateway.
```

File: docs/data-sources/nat_private_gateways.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  + **Small**: Small type, which supports up to `20` rules, `200 Mbit/s` bandwidth, `20,000` PPS and `2,000` SNAT
+    connections.
+  + **Medium**: Medium type, which supports up to `50` rules, `500 Mbit/s` bandwidth, `50,000` PPS and `5,000` SNAT
+    connections.
+  + **Large**: Large type, which supports up to `200` rules, `2 Gbit/s` bandwidth, `200,000` PPS and `20,000` SNAT
+    connections.
+  + **Extra-Large**: Extra-large type, which supports up to `500` rules, `5 Gbit/s` bandwidth, `500,000` PPS and
+    `50,000` SNAT connections.
+
+* `description` - (Optional, String) Specifies the description of the private NAT gateways.
```
Comment (Lance52259) [inline docs/data-sources/nat_private_gateways.md]: take this parameter to the attribute list.

File: docs/data-sources/nat_private_gateways.md (hunk 2 above)
Comment (ruwenqiang123) [reply docs/data-sources/nat_private_gateways.md]: have changed

File: docs/data-sources/nat_private_gateways.md (hunk 1 above)
Comment (ruwenqiang123) [reply docs/data-sources/nat_private_gateways.md]: have changed

File: docs/data-sources/nat_private_gateways.md (hunk 3 above)
Comment (ruwenqiang123) [reply docs/data-sources/nat_private_gateways.md]: have changed

**Pull request merged**